### PR TITLE
Adds Records library

### PR DESCRIPTION
**Pull request by Kitsunemitsu**
## About The Pull Request
Adds in a new ghost role for reading records, and later working on the library.

## Why It's Good For The Game
I think something like this could be cute

## Changelog
:cl:
add: Added Records library and ghost role
/:cl:

### CODE REVIEW

File: code/modules/paperwork/records/recordsfiling.dm [hunk 1]
```diff
@@ -111,3 +111,30 @@
...
+
+/*
+ * Lore Cabinet
+ */
+
+//Zayin
+/obj/structure/filingcabinet/zayinlore
+	name = "zayin supplimentary information cabinet"
+	icon_state = "employmentcabinet"
+	var/virgin = TRUE
```
Comment (EgorDinamit) [inline code/modules/paperwork/records/recordsfiling.dm:123]: This isn't used anywhere.

File: code/modules/paperwork/records/recordsfiling.dm [hunk 2]
```diff
@@ -111,3 +111,30 @@
...
+	var/list/queue = subtypesof(/obj/item/paper/fluff/lore/zayin)
+	for(var/sheet in queue)
+		new sheet(src)
+
+
+//Waw
+/obj/structure/filingcabinet/wawlore
+	name = "waw supplimentary information cabinet"
+	icon_state = "employmentcabinet"
+	var/virgin = TRUE
```
Comment (EgorDinamit) [inline code/modules/paperwork/records/recordsfiling.dm:135]: Same here.

File: code/modules/paperwork/records/recordsfiling.dm [hunk 3]
```diff
@@ -111,3 +111,30 @@
...
+ * Lore Cabinet
+ */
+
+//Zayin
+/obj/structure/filingcabinet/zayinlore
+	name = "zayin supplimentary information cabinet"
+	icon_state = "employmentcabinet"
+	var/virgin = TRUE
+
+/obj/structure/filingcabinet/zayinlore/proc/fillCurrent()
```
Comment (EgorDinamit) [inline code/modules/paperwork/records/recordsfiling.dm:125]:
```suggestion
/obj/structure/filingcabinet/zayinlore/Initialize()
```
Either that, or actually add a proc call for the FillCurrent().

File: code/game/objects/structures/ghost_role_spawners.dm [hunk 4]
```diff
@@ -944,3 +944,38 @@
...
+	desc = "A humming sleeper with a silhouetted occupant inside. It's just a front, stored underground are many librarians."
+	mob_name = "Records Librarian"
+	icon = 'icons/obj/machines/sleeper.dmi'
+	icon_state = "sleeper"
+	outfit = /datum/outfit/librarian
+	short_desc = "A place for players to read up on lore and information while not affecting the actual round \
+	When you are done, Cryo and respawn to join the game."
+	assignedrole = "Lobotomy Corporation Records Librarian"
+
+/obj/effect/mob_spawn/human/library/Destroy()
```
Comment (EgorDinamit) [inline code/game/objects/structures/ghost_role_spawners.dm:965]: Now that I think about it, why are you doing this all? Ghost role spawners have variables for infinite "spawn slots" already.

Comment (Kitsunemitsu) [reply code/game/objects/structures/ghost_role_spawners.dm:965]: I didn't see it when I was working on the intern, so I just winged it. This works fine but I will fix

File: code/modules/paperwork/records/recordsfiling.dm (hunk 3 above)
Comment (Kitsunemitsu) [reply code/modules/paperwork/records/recordsfiling.dm]: It does call fill current, first time you attack hand, and it's why the virgin var is needed, also removes a lot of "Spawn 200 papers inside cabinets" on startup
Using initialize causes integration errors without giving me logs.

**Review by dsmith328 — APPROVED** [review]

**Pull request merged**